### PR TITLE
Moved pathOrContent to separate package.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

This removes some complexity, plus it allows other projects to use this (e.g https://github.com/thanos-io/thanosbench)

### CODE REVIEW

File: docs/components/bucket.md [hunk 1]
```diff
@@ -101,23 +103,23 @@ Flags:
...
+      --tracing.config-file=<file-path>
                                Path to YAML file that contains tracing
                                configuration. See format details:
                                https://thanos.io/tracing.md/#configuration
-      --tracing.config=<tracing.config-yaml>
+      --tracing.config=<content>
                                Alternative to 'tracing.config-file' flag.
-                               Tracing configuration in YAML. See format
-                               details:
+                               Content of YAML file that contains tracing
```
Comment (FUSAKLA) [inline docs/components/bucket.md]: nit: I understand that it's composed of the generic prefix and flag help but the resulting wording "content of file that contains" sounds bit weird to me.  "Content of YAML file with tracing configuration." maybe? 
But perhaps it is just me.

File: pkg/extflag/pathorcontent.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+type CmdClause interface {
+	Flag(name, help string) *kingpin.FlagClause
+}
+
+// RegisterPathOrContent registers PathOrContent flag in kingpinCmdClause.
+func RegisterPathOrContent(cmd CmdClause, flagName string, help string, required bool) *PathOrContent {
+	fileFlagName := fmt.Sprintf("%s-file", flagName)
+	contentFlagName := flagName
+
+	fileHelp := fmt.Sprintf("Path to %s", help)
```
Comment (FUSAKLA) [inline pkg/extflag/pathorcontent.go]:
```suggestion
	fileHelp := fmt.Sprintf("Path to %s (has priority over the content flag)", help)
```
Might be worth to add the priority somewhere but not sure if this is the right way/place.

File: docs/components/bucket.md (hunk 1 above)
Comment (bwplotka) [reply docs/components/bucket.md]: happy to change

File: pkg/extflag/pathorcontent.go (hunk 2 above)
Comment (bwplotka) [reply pkg/extflag/pathorcontent.go]: I will add

**Pull request merged**